### ISSUE TITLE
soapysdr and soapyrtlsdr try LoadLibrary call with badpath

### ISSUE THREAD

**Issue by chrisjohgorman** [msys2/MINGW-packages#21163]
### Description / Steps to reproduce the issue

I see this when trying to load the library from `gnuradio`.  There are a lot of packages required to built to get `gnuradio` on your system to see what I see and I do not have a minimal working example.  Anyway to install `gnuradio` the necessary packages are available at my fork of `MINGW-packages`.  It's `https://github.com/chrisjohgorman/MINGW-packages.git` on the branch `new-pkgbuilds`.   I hope it's not necessary to go through the trouble of building `gnuradio` to fix this.

Any way, when trying to use a `soapyrtlsdr` source block in gnuradio, I get the following error.
```
[ERROR] SoapySDR::loadModule(C:\msys64\mingw64/lib/SoapySDR/modules0.8/librtlsdrSupport.dll)
  LoadLibrary() failed: The specified procedure could not be found.
```
The path names are mixed.   The prefix path separator is Win32 and the suffix path separator is Unix.

### Expected behavior

I expect that the path separator should be either Unix or Win32.  (And the module should load.)

### Actual behavior

The actual behaviour is an error message and the module fails to load.  This causes the block to fail under `gnuradio`.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-22631

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

If I figure out how to fix this myself, I will make a change, patch and PR

**Comment by mmuetzel** [comment]
The mix of forward slashes and backslashes is probably a red herring.

Afaict, the error message "The specified procedure could not be found." means that it found the library itself. But it is missing one of its dependencies.
Maybe, an issue with the order in which the `.dll`s appear in your path?

According to Dependencies, `librtlsdrSupport.dll` depends on:
* `C:\msys64\mingw64\bin\libgcc_s_seh-1.dll`
* `C:\WINDOWS\system32\kernel32.dll`
* `C:\WINDOWS\system32\MSVCRT.dll`
* `C:\msys64\mingw64\bin\librtlsdr.dll`
* `C:\msys64\mingw64\bin\libstdc++-6.dll`
* `C:\msys64\mingw64\bin\libSoapySDR.dll`

What does `echo $PATH` return for you in the environment where you see that error?

**Comment by chrisjohgorman** [comment]
My path is ...
```
# echo $PATH
/mingw64/bin:/mingw64/bin/site_perl/5.38.2:/mingw64/bin/vendor_perl:/mingw64/bin/core_perl:/usr/local/bin:/usr/bin:/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/home/chris/bin
```
and I have all the required libs ...
```
# ntldd librtlsdrSupport.dll
        libgcc_s_seh-1.dll => C:\msys64\mingw64\bin\libgcc_s_seh-1.dll (0x0000015dda490000)
        KERNEL32.dll => C:\WINDOWS\SYSTEM32\KERNEL32.dll (0x0000015dda4d0000)
        msvcrt.dll => C:\WINDOWS\SYSTEM32\msvcrt.dll (0x0000015ddacf0000)
        librtlsdr.dll => C:\msys64\mingw64\bin\librtlsdr.dll (0x0000015dda490000)
        libstdc++-6.dll => C:\msys64\mingw64\bin\libstdc++-6.dll (0x0000015dda4d0000)
        libSoapySDR.dll => C:\msys64\mingw64\bin\libSoapySDR.dll (0x0000015dda4d0000)
```
Now to try to find out if it's failing to get that pointer.

**Comment by chrisjohgorman** [comment]
Forgot to add my thanks Markus, and Biswapriyo.

**Comment by chrisjohgorman** [comment]
Well, I don't exactly know what was wrong here, but a rebuild of the packages, fixes this error for me.  I believe the rebuild of soapysdr was the one that fixed it, as it had a different size when rebuilt, but I rebuilt both before testing, so I can't be sure.  Anyway thanks Markus and Biswapriyo for your help in identifying and fixing this.  All the best.

**Comment by chrisjohgorman** [comment]
Fixed.

**Comment by chrisjohgorman** [comment]
I think I closed this in haste.  Without a rebuild users of `gnuradio` cannot use the `soapyrtlsdr` source block.  Is it possible to rebuild this package?   It's build date is June 2023.  I am willing to make a PR to do this if it's ok with @antoniovazquezblanco .

**Comment by chrisjohgorman** [comment]
Upon further review and testing `mingw-w64-x86_64-soapyrtlsdr` needs to be rebuilt.  In December 2023, `mingw-w64-x86_64-rtl-sdr` was updated to 2.0.1.  The package `soapyrtlsdr` depends on `rtl-sdr`, but `soapyrtlsdr` hasn't been updated since June 2023.   Since this is a simple fix, I've made a PR for this problem.  https://github.com/msys2/MINGW-packages/pull/21515

I have tested the artifacts from `mingw64` and `ucrt64` and both new packages work.  I have neglected to test `clang64` as we don't have a working `gnuradio` for it at the moment.
